File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    ArrayList<Integer> values = new ArrayList<Integer>();
    public void traverse(TreeNode node){
        if (node != null){
            traverse(node.left);
            traverse(node.right);
            values.add(node.val);
        }
    }
    
    public List<Integer> postorderTraversal(TreeNode root) {
        traverse(root);
        return values;
    }
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    
    def traverse(self, node: TreeNode) -> None:
        if node:
            self.traverse(node.left)
            self.traverse(node.right)
            self.values.append(node.val)
        

    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.values = []
        self.traverse(root)
        return self.values

